### PR TITLE
Change: Colour -> Color (Filenames and file content)

### PR DESCRIPTION
**Pull request by davidlamhauge**
This PR affects only the development side.
I've changed 'colour' to 'color' in all files, and renamed the class 'ColourRef' to 'ColorRef'. There seemed to consensus on 'color' last time we discussed it, and it is consistent with Qt, who has QColor, QColorDialog etc.
I've used Qt Creators Search and Replace. When I search now, there are still 22 strings that contain 'olour', and all of them are *.qm files in 'translations'. Is this a problem?

### CODE REVIEW

**Comment by chchwy** [comment]
Anything regarding backward compatibility should not be changed.

**Comment by davidlamhauge** [comment]
I have re-instated 'colour' five places, to avoid any problems with backwards compatibility.

File: core_lib/src/structure/object.cpp [hunk 1]
```diff
@@ -419,13 +419,13 @@ void Object::exportPalettePencil(QFile& file)
     doc.appendChild(root);
     for (int i = 0; i < mPalette.size(); i++)
     {
-        ColourRef ref = mPalette.at(i);
-        QDomElement tag = doc.createElement("Colour");
+        ColorRef ref = mPalette.at(i);
+        QDomElement tag = doc.createElement("Color");
```
[the remark this answers is not in the document]
Comment (davidlamhauge) [reply core_lib/src/structure/object.cpp:423]: I've kept the 'Color' here, since the import code doesn't check whether it is 'Colour' or 'Color'.

**Comment by chchwy** [comment]
Alright. Can you please also revert 2 things:
1. `catch.hpp` <= it's a 3rd party library.
2.  all the `*.ts` file, there are auto-generated files so they will be overwritten next time updating translations.

File: tests/src/catch.hpp [hunk 2]
```diff
@@ -219,7 +219,7 @@ namespace Catch {
 #  endif
 
 // Universal Windows platform does not support SEH
-// Or console colours (or console at all...)
+// Or console colors (or console at all...)
```
[the remark this answers is not in the document]
Comment (davidlamhauge) [reply tests/src/catch.hpp]: Reverted...

**Comment by davidlamhauge** [comment]
Done changes as requested

**Pull request merged**